ast-nodes/expressions/operators/neg-operator.dart

```diff
@@ -25,6 +25,7 @@ class NegOperator extends UnaryRelation implements Primary {
   }
 
   void checkSemantics() {
-    // TODO: implement
+    this.operand.checkSemantics();
+    ensureType(this.operand, this.resultType);
```

Comment:
You are the one that has to set the resulting type, you cannot use it. And the same mistake with `ensureType` as in one of the previous commits.